consensus/core/src/transaction.rs

```diff
@@ -18,19 +19,27 @@ const MAX_PENDING_TRANSACTIONS: usize = 2_000;
 
 const MAX_CONSUMED_TRANSACTIONS_PER_REQUEST: u64 = 5_000;
 
+/// The guard acts as an acknowledgment mechanism for the inclusion of the transaction to a block.
+/// When the guard is dropped then the transaction has been acknowledged.
+pub(crate) struct TransactionGuard {
+    pub transaction: Transaction,
+    // implicitly dropping the sender acknowledges the transaction
+    _included_in_block_ack: oneshot::Sender<()>,
+}
```

Comment:
I went the the "guard" approach with the implicit acknowledgment when dropped as the use is quite straightforward and doesn't need much handling from our end. Another approach by having a dedicated `acknowledge` method was considered, but I thought it would be more error prone in case where a transaction would be included but never explicitly acknowledged and dropped - that could create false negatives on the submission side which would be quite bad. Also as the block proposal path doesn't handle any error retires - but rather panic on error - we don't really need additional error semantics.

Reply:
Sounds good. But I think there should be a way to return an error to Sui in case consensus is shutting down. Currently the caller cannot tell the difference between transaction included and the consensus shutting down. If Sui relies on this status to return to fullnode, there could be more confusions. I think receiving on the receiver should only confirm inclusion in a block, and dropping sender should be reserved for shut down? In future we may even want to send the block ref back to Sui.

Reply:
The consensus shutdown is a good point. The guarantees are still slim - I mean consensus might still immediately shutdown after the acknowledgment. I guess that's still good enough as the block in the best case will get re-broadcasted after the restore or implicitly served via a fetching approach. I'll refactor the code and make this more explicit  via an acknowledge method - and we'll need to make sure that we don't pre-maturely drop the guard.